ShopParser/ComponentsParser/ProcessorParser.cs

```diff
@@ -0,0 +1,79 @@
+﻿using ItemLibrary;
+using OpenQA.Selenium;
+using OpenQA.Selenium.Chrome;
+using OpenQA.Selenium.Support.UI;
+using Parsing;
+using PathLibrary;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace WebParser.ComponentsParser
+{
+    public class ProcessorParser
+    {
+        private readonly string _url = "https://www.gpuskin.com/CPU-Database";
+        private Lazy<ChromeDriver> _driver;
+
+        public ProcessorParser()
+        {
+            var options = new ChromeOptions();
+            options.AddArguments("--headless");
+            _driver = new Lazy<ChromeDriver>(() => new ChromeDriver(MainPath.GetShopParserPath(), options));
+
+        }
+
+
+        public Processor ParseProcessor (string model)
+        {
+            _driver.Value.Navigate().GoToUrl(_url);
+            _driver.Value.FindElement(By.Name("SearchRecords")).SendKeys(model);
+
+            DefaultWait<IWebDriver> fluentWait = new DefaultWait<IWebDriver>(_driver.Value);
+            fluentWait.Timeout = TimeSpan.FromSeconds(5);
+            fluentWait.PollingInterval = TimeSpan.FromMilliseconds(250);
+            fluentWait.IgnoreExceptionTypes(typeof(NoSuchElementException));
+            IWebElement searchResult = fluentWait.Until(x => x.FindElement(By.Id("searchButton")));
+
+            _driver.Value.FindElement(By.Id("searchButton")).Click();
+
+
+            var button = _driver.Value.FindElements(By.CssSelector("a.btn.btn-default"));
+
+             var link = button[0].GetAttribute("href");
+
+            _driver.Value.Navigate().GoToUrl(link);
+
+     
+            var table = _driver.Value.FindElements(By.TagName("td"));
+
+            Processor processor = new Processor ();
+
+            processor.Model = _driver.Value.FindElement(By.ClassName("text-center"))
+                .GetAttribute("innerText").Split(new[] { '\r', '\n' }).FirstOrDefault();
+
+            for (int i = 0; i < table.Count; i++)
+            {
+                if (table[i].GetAttribute("innerHTML").Contains("Cache L3"))
+                {
+                    processor.Cache = table[i + 1].GetAttribute("innerHTML").ParseInt();
+                }
+            }
+            processor.MinCores = _driver.Value.FindElement(By.CssSelector("#fh5co-programs-section > div > form > div:nth-child(6) > div > table > tbody > tr:nth-child(2) > td:nth-child(2) > div > div:nth-child(2) > div:nth-child(1)"))
+                .GetAttribute("innerHTML").ParseInt();
+
+            ResetDriver();
```

Comment:
Processor Name (a.k.a type) should be set with method SetName() in this parser before returning the processor?

Reply:
waiting for your method SetType to be updated, then will add it in parser